packages/core/core/src/AssetGraph.js

```diff
@@ -280,15 +321,29 @@ export default class AssetGraph extends Graph<AssetGraphNode> {
     return res;
   }
 
-  traverseAssets(
-    visit: GraphTraversalCallback<Asset, AssetGraphNode>,
-    startNode: ?AssetGraphNode
-  ): ?AssetGraphNode {
-    return this.traverse((node, ...args) => {
-      if (node.type === 'asset') {
-        return visit(node.value, ...args);
+  getAncestorDependencies(asset: Asset): Array<IDependency> {
+    let node = this.getNode(asset.id);
+    if (!node) {
```

Comment:
should we be throwing in situations like these?

Reply:
maybe, but in this case if you asked for ancestor dependencies of an entry I think it's better to return an empty array than throw.